pyvista/plotting/renderers.py

```diff
@@ -312,13 +312,13 @@ def index_to_loc(self, index):
 
         Returns
         -------
-        numpy.ndarray
+        numpy.ndarray or numpy.int64
             2D location on the plotting grid.
         """
         if not isinstance(index, (int, np.integer)):
             raise TypeError('"index" must be a scalar integer.')
         if len(self.shape) == 1:
-            return index
+            return np.int64(index)
```

Comment:
I'm wondering if we should change this to
```suggestion
            return np.intp(index)
```
(the return type annotation is probably better left as `int64` for end users).

[To quote Ralf Gommers](https://github.com/numpy/numpy/issues/9464#issuecomment-2185930655):
> This has been addressed in NumPy 2.0 by changing the default integer type to `intp`, which has the practical consequence of changing the default on 64-bit Windows from `int32` to `int64`.

[To quote the 2.0 release notes](https://numpy.org/doc/stable/release/2.0.0-notes.html#redefinition-of-np-intp-np-uintp-almost-never-a-change):
> Due to the actual use of [`np.intp`/`np.uintp`] almost always matching the use of `size_t`/`Py_ssize_t` this is now the definition in C. Previously, it matched `intptr_t` and `uintptr_t` which would often have been subtly incorrect. This has no effect on the vast majority of machines since the size of these types only differ on extremely niche platforms.

"Extremely niche platforms" are unlikely to be our userbase, right?

It can also make sense to go with explicit `int64`, but then we must ensure that all code paths return that, and that the array case has the same dtype.

Reply:
Although the "size of these types only differ" might have referred to `size_t` vs `intptr_t` rather than `int64` vs `intp` so that quote might be a weaker indication for my point.

Reply:
Given your quote it sounds like the preference should be `np.intp`, although I'm not fully grokking the practical difference between `np.int_` and `np.intp`.  Given `np.intp` has something to do with pointer sizes, and this parameter is an index value, it makes logical sense here.

I highly suspect that there will be no impact to any of the choices here.